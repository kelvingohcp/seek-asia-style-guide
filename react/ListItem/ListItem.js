import styles from './ListItem.less';
import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Text from '../Text/Text';

export default function ListItem({
  title,
  value,
  icon,
  badge,
  isCompact,
  hasHoverState,
  noShadow
}) {
  return (
    <div
      className={classnames({
        [styles.root]: true,
        [styles.isCompact]: isCompact,
        [styles.hasHoverState]: hasHoverState,
        [styles.noShadow]: noShadow
      })}
    >
      {icon && <div className={classnames(styles.listItemIcon)}>{icon}</div>}
      <div className={classnames(styles.listItemBody)}>
        {title && (
          <Text
            light
            intimate
            baseline={false}
            className={classnames(styles.listItemTitle)}
          >
            {title}
          </Text>
        )}
        <div className={classnames(styles.listItemValue)}>
          <Text
            strong
            loud
            baseline={false}
            className={classnames(styles.displayInline)}
          >
            {value}
          </Text>
          {badge && (
            <span>
              {' '}
              <span className={classnames(styles.listItemBadge)}> {badge}</span>
            </span>
          )}
        </div>
      </div>
    </div>
  );
}

ListItem.propTypes = {
  title: PropTypes.string,
  value: PropTypes.string.isRequired,
  icon: PropTypes.function,
  badge: PropTypes.function,
  isCompact: PropTypes.bool,
  hasHoverState: PropTypes.bool,
  noShadow: PropTypes.bool
};

ListItem.defaultProps = {
  isCompact: false,
  hasHoverState: false,
  noShadow: false
};
