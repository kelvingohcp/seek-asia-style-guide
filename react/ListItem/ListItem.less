@import (reference) "~seek-asia-style-guide/theme";

@list-item-v-padding: @grid-base * 3;
@list-item-h-padding: @grid-base * 4;
@list-item-line-height: 20px;
@list-item-bgc-hover: #eef8fc;
@list-item-icon-margin-top: @grid-base / 2;
@list-item-icon-padding-left: @grid-base * 3;

.root {
  align-items: flex-start;
  background-color: @saWhite;
  border-radius: 0.25rem;
  box-shadow: 0 1px 1px @saGrey5;
  color: @saGrey1;
  display: flex;
  font-size: 16px;
  justify-content: space-between;
  line-height: @list-item-line-height;
  padding-top: @list-item-v-padding;
  padding-bottom: @list-item-v-padding;
  @media @desktop {
    max-width: 50vw;
  }
  &:hover {
    background-color: @list-item-bgc-hover;
  }
}

.listItemBody {
  display: flex;
  flex: 1 0 0;
  flex-direction: column;
  padding-left: @list-item-h-padding;
  padding-right: @list-item-h-padding;
}

.listItemTitle {
  color: @saGrey2;
  display: block;
  font-size: 0.875em;
}

.listItemIcon {
  line-height: inherit;
  margin-top: @list-item-icon-margin-top;
  padding-left: @list-item-icon-padding-left;
}

.listItemValue {
  flex: 1 0 0;
}
