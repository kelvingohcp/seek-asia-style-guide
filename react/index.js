export { default as StyleGuideProvider } from './StyleGuideProvider/StyleGuideProvider';
export { default as SeekApp } from './StyleGuideProvider/StyleGuideProvider'; // Alias for backwards compatibility

// Components
export { default as Alert } from './Alert/Alert';
export { default as Button } from './Button/Button';
export { default as FilterButton } from './FilterButton/FilterButton';
export { default as ButtonGroup } from './ButtonGroup/ButtonGroup';
export { default as ListItem } from './ListItem/ListItem';
export { default as Loader } from './Loader/Loader';
export { default as Logo } from './Logo/Logo';
export { default as LogoRainbow } from './LogoRainbow/LogoRainbow';
export { default as Rating } from './Rating/Rating';
export { default as NavigationBar } from './NavigationBar/NavigationBar';
export { default as JobCard, trackLinkType } from './JobCard/JobCard';
export { default as JobCardLoader } from './JobCardLoader/JobCardLoader';
export { default as CardLoader } from './CardLoader/CardLoader';
export { default as CardLoadingBar } from './CardLoadingBar/CardLoadingBar';
export { default as Header } from './Header/Header';
export { default as ShowMore } from './ShowMore/ShowMore';
export { default as TitledCard } from './TitledCard/TitledCard';
export { default as Icon } from './Icon/Icon';
export { default as DropdownList } from './DropdownList/DropdownList';
export { default as Tag } from './Tag/Tag';
export { default as Badge } from './Badge/Badge';

// Typography
export { default as Critical } from './Critical/Critical';
export { default as Paragraph } from './Paragraph/Paragraph';
export { default as Positive } from './Positive/Positive';
export { default as Secondary } from './Secondary/Secondary';
export { default as Strong } from './Strong/Strong';
export { default as Text } from './Text/Text';
export { default as TextLink } from './TextLink/TextLink';

// Layout
export { default as AsidedLayout } from './AsidedLayout/AsidedLayout';
export { default as Card } from './Card/Card';
export { default as CardGroup } from './CardGroup/CardGroup';
export { default as Columns } from './Columns/Columns';
export { default as Hidden } from './Hidden/Hidden';
export { default as PageBlock } from './PageBlock/PageBlock';
export { default as Section } from './Section/Section';

// Icons
export { default as CriticalIcon } from './CriticalIcon/CriticalIcon';
export { default as CrossIcon } from './CrossIcon/CrossIcon';
export { default as ErrorIcon } from './ErrorIcon/ErrorIcon';
export { default as FieldOfStudyIcon } from './FieldOfStudyIcon/FieldOfStudyIcon';
export { default as FlagHKIcon } from './FlagHKIcon/FlagHKIcon';
export { default as FlagIDIcon } from './FlagIDIcon/FlagIDIcon';
export { default as FlagMYIcon } from './FlagMYIcon/FlagMYIcon';
export { default as FlagPHIcon } from './FlagPHIcon/FlagPHIcon';
export { default as FlagSGIcon } from './FlagSGIcon/FlagSGIcon';
export { default as FlagTHIcon } from './FlagTHIcon/FlagTHIcon';
export { default as FlagVNIcon } from './FlagVNIcon/FlagVNIcon';
export { default as HelpIcon } from './HelpIcon/HelpIcon';
export { default as InfoIcon } from './InfoIcon/InfoIcon';
export { default as JobFunctionIcon } from './JobFunctionIcon/JobFunctionIcon';
export { default as JobInvitationIcon } from './JobInvitationIcon/JobInvitationIcon';
export { default as MoreIcon } from './MoreIcon/MoreIcon';
export { default as ProfileIcon } from './ProfileIcon/ProfileIcon';
export { default as StarIcon } from './StarIcon/StarIcon';
export { default as TickCircleIcon } from './TickCircleIcon/TickCircleIcon';

// Fields
export { default as Autosuggest } from './Autosuggest/Autosuggest';
export { default as Checkbox } from './Checkbox/Checkbox';
export { default as ClearField } from './ClearField/ClearField';
export { default as Dropdown } from './Dropdown/Dropdown';
export { default as EmailField } from './EmailField/EmailField';
export { default as MonthPicker } from './MonthPicker/MonthPicker';
export { default as SlideToggle } from './SlideToggle/SlideToggle';
export { default as Textarea } from './Textarea/Textarea';
export { default as TextField } from './TextField/TextField';

// Accessibility
export { default as ScreenReaderOnly } from './ScreenReaderOnly/ScreenReaderOnly';
export { default as ScreenReaderSkipLink } from './ScreenReaderSkipLink/ScreenReaderSkipLink';
export { default as ScreenReaderSkipTarget } from './ScreenReaderSkipTarget/ScreenReaderSkipTarget';

// Constants
export { default as Constants } from './Constants/Constants';
