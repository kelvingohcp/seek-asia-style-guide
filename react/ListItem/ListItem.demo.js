import React from 'react';
import ListItem from './ListItem';
import Badge from '../Badge/Badge';
import * as sketch from './ListItem.sketch';
import { HeartIcon, StarIcon } from 'seek-asia-style-guide/react';
export default {
  route: '/listItem',
  title: 'List Item',
  component: ListItem,
  sketch,
  initialProps: {
    value: 'Description'
  },
  options: [
    {
      label: 'Title',
      type: 'radio',
      states: [
        {
          label: 'No Title',
          transformProps: props => props
        },
        {
          label: 'Have Title',
          transformProps: props => ({
            ...props,
            title: 'Job Posted Date',
            value: '10 May 2018'
          })
        }
      ]
    },
    {
      label: 'Icon',
      type: 'radio',
      states: [
        {
          label: 'No Icon',
          transformProps: props => props
        },
        {
          label: 'Heart Icon',
          transformProps: props => ({
            ...props,
            icon: <HeartIcon />
          })
        },
        {
          label: 'Star Icon',
          transformProps: props => ({
            ...props,
            icon: <StarIcon />
          })
        }
      ]
    },
    {
      label: 'with Badge',
      type: 'checklist',
      states: [
        {
          label: 'with Badge',
          transformProps: props => ({
            ...props,
            badge: <Badge label="Badge" color="default" isBold />
          })
        }
      ]
    },
    {
      label: 'Has hover state',
      type: 'checklist',
      states: [
        {
          label: 'Has hover state',
          transformProps: props => ({
            ...props,
            hasHoverState: true
          })
        }
      ]
    },
    {
      label: 'No shadow',
      type: 'checklist',
      states: [
        {
          label: 'No shadow',
          transformProps: props => ({
            ...props,
            noShadow: true
          })
        }
      ]
    }
  ]
};
