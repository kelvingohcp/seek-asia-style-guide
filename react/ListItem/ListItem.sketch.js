import React from 'react';
import mapKeys from 'lodash/mapKeys';
import ListItem from './ListItem';
import { IconEducation } from 'seek-asia-style-guide/react';

const badge = {
  label: 'Badge'
};

export const listItem = {

  'value': <ListItem value='Description' />,
  'value, hover': <ListItem value='Description' hasHoverState />,
  'value, no shadow': <ListItem value='Description' noShadow />,
  'title, value': <ListItem title='Title' value="Description" />,
  'title, value, icon': <ListItem title='Qualification' value='Degree' icon={<IconEducation />} />,
  'title, value, icon, compact': <ListItem title='Qualification' value='Degree' compact icon={<IconEducation />} />,
  'title, value, icon, badge': <ListItem badge={{ ...badge }} icon={<IconEducation />} title="Job Posted Date" value="10 May 2018" />,
  'title, value, icon, badge(diff color)': <ListItem badge={{ ...badge, color: 'progressing' }} icon={<IconEducation />} title="Job Posted Date" value="10 May 2018" />,
  'title, value, icon, hasHoverState, badge(diff color)': <ListItem badge={{ ...badge, color: 'progressing' }} icon={<IconEducation />} title="Job Posted Date" value="10 May 2018" hasHoverState />
};

// Export text styles as symbols
// (at least until Sketch libraries support shared text styles)
export const symbols = mapKeys(listItem, (value, key) => `ListItem/${key}`);
