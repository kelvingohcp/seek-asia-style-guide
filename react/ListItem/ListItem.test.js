import React from 'react';
import { shallow } from 'enzyme';
import ListItem from './ListItem';
import { Icons } from 'seek-asia-style-guide/react';
import styles from './ListItem.less';

describe('ListItem:', () => {
  it('should render value', () => {
    const wrapper = shallow(<ListItem value="testValue" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render title value', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom left padding', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" leftPadding={8} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom left padding with light font weight', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" leftPadding={8} descriptionProps={{ light: true }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom left padding with light font weight and secondary text', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" leftPadding={8} descriptionProps={{ light: true, secondary: true }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render custom left padding with light font weight, secondary text and onclick function', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" leftPadding={8} descriptionProps={{ light: true, secondary: true }} onClick={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render icon', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" icon={<Icons iconName="education" />} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render icon in compact mode', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" icon={<Icons iconName="education" />} compact />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render icon in hasHoverState mode', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" icon={<Icons iconName="education" />} hasHoverState />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render icon with badge with hasHoverState and compact mode', () => {
    const wrapper = shallow(<ListItem value="testValue" title="testTitle" icon={<Icons iconName="education" />} hasHoverState compact badge={{ color: 'default', isBold: true, label: 'badge' }} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render with className', () => {
    const wrapper = shallow(<ListItem value="testValue" className={styles.root} />);
    expect(wrapper).toMatchSnapshot();
  });
});
