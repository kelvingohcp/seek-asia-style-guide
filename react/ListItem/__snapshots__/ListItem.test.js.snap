// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem: should render custom left padding 1`] = `
<div
  className="root"
>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={true}
      regular={false}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render custom left padding with light font weight 1`] = `
<div
  className="root"
>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={true}
      regular={false}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        light={true}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render custom left padding with light font weight and secondary text 1`] = `
<div
  className="root"
>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={true}
      regular={false}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        light={true}
        secondary={true}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render custom left padding with light font weight, secondary text and onclick function 1`] = `
<div
  className="root"
>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={true}
      regular={false}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        light={true}
        secondary={true}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render icon 1`] = `
<div
  className="root"
>
  <div
    className="listItemIcon"
  >
    <Education icon />
  </div>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={true}
      regular={false}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render icon in compact mode 1`] = `
<div
  className="root"
>
  <div
    className="listItemIcon"
  >
    <Education icon />
  </div>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={false}
      regular={true}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={true}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render icon in hasHoverState mode 1`] = `
<div
  className="root hasHoverState"
>
  <div
    className="listItemIcon"
  >
    <Education icon />
  </div>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={true}
      regular={false}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render icon with badge with hasHoverState and compact mode 1`] = `
<div
  className="root hasHoverState"
>
  <div
    className="listItemIcon"
  >
    <Education icon />
  </div>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={false}
      regular={true}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={true}
        semiStrong={false}
      >
        testValue
      </Text>
      <span
        className="listItemBadge"
      >
        <Badge
          color="default"
          isBold={true}
          label="badge"
        />
      </span>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render title value 1`] = `
<div
  className="root"
>
  <div>
    <Text
      baseline={false}
      className="listItemTitle"
      intimate={true}
      light={true}
      regular={false}
    >
      testTitle
    </Text>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render value 1`] = `
<div
  className="root"
>
  <div>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;

exports[`ListItem: should render with className 1`] = `
<div
  className="root"
>
  <div>
    <div
      className="listItemValue"
    >
      <Text
        baseline={false}
        className="displayInline"
        intimate={false}
        semiStrong={false}
      >
        testValue
      </Text>
    </div>
  </div>
</div>
`;
